src/main/java/org/apache/hadoop/mapred/MesosScheduler.java

```diff
@@ -181,15 +183,36 @@ public synchronized void start() throws IOException {
     String master = conf.get("mapred.mesos.master", "local");
 
     try {
-      FrameworkInfo frameworkInfo = FrameworkInfo
+
+      FrameworkInfo.Builder frameworkInfoBuilder = FrameworkInfo
         .newBuilder()
-        .setUser("") // Let Mesos fill in the user.
+        .setUser(conf.get("mapred.mesos.framework.user","")) // Let Mesos fill in the user.
         .setCheckpoint(conf.getBoolean("mapred.mesos.checkpoint", false))
         .setRole(conf.get("mapred.mesos.role", "*"))
-        .setName("Hadoop: (RPC port: " + jobTracker.port + ","
-                 + " WebUI port: " + jobTracker.infoPort + ")").build();
-
-      driver = new MesosSchedulerDriver(this, frameworkInfo, master);
+        .setName(conf.get("mapred.mesos.framework.name","Hadoop: (RPC port: " + jobTracker.port + ","
+                 + " WebUI port: " + jobTracker.infoPort + ")"));
+      Credential credential=null;
+
+      String frameworkPrincipal=conf.get("mapred.mesos.framework.principal");
+      if (frameworkPrincipal!=null){
```

Comment:
Would you mind just tidying up this block of code you've added to be consistent with the rest of the file (basically just some additional spaces around your `if` statements and `=` signs.

Thanks!